oop2-encapsulate-inherit.js

```diff
@@ -6,28 +6,342 @@
+  }
+
+  getBooks() {
+    return this.#books;
+  }
+
+  setName(name) {
+    this.#name = name;
+  }
+
+  setBooks(books) {
+    this.#books = books;
+  }
+}
+
+class Publisher {
+  #authors;
+  #books;
+
+  constructor() {}
+
+  getAuthors() {
+    return this.#authors;
+  }
 
+  getBooks() {
+    return this.#books;
+  }
 
+  setAuthors(authors) {
+    this.#authors = authors;
+  }
 
+  setBooks(books) {
+    this.#books = books;
+  }
+}
 
+class Review {
+  #rating;
+  #user;
 
+  constructor() {}
 
+  getRating() {
+    return this.#rating;
+  }
 
+  getUser() {
+    return this.#user;
+  }
+
+  setRating() {
+    this.#rating = rating;
+  }
+
+  setUser() {
+    this.#user = user;
+  }
+}
+
+/*
 2. Create the following classes:
     - Umbrella class.  It should have a an organization name.  It should only have a single instance.  Umbrella is the "Parent Organization" e.g. Allegis Group
     - Company class. It should have a compnay name and # of employees.  Create at least 3 different companies that are children of the Umbrella Organization.  (e.g. TEKsystems, Aerotek, etc.)
     - Site class. It should have the name of the company, and the location of the site.  It is a child class that inherits from the Company class.  Create 3 sites for each company.
     - Employee class. It should have a employee name, job title, and salary properties. Create 10 employee instances: 1 CEO, 1 Manager, 1 Secretary, 2 Engineers, 1 Financial Officer, 1 Janitor, 1 Marketer, 1 HR Personnel, 1 Lawyer.  Feel free to add other employee roles at your own discretion.
     
     - For each class, add 2 properties and 2 methods to each.  The properties and methods should make sense, considering what the classes are supposed to be representing.
+*/
+
+class Umbrella {
+  name;
+  #companies = [];
+
+  constructor(name) {
+    this.name = name;
+  }
+
+  addNewCompany(company) {
+    this.#companies.push(company);
+    console.log(`${this.name} has added ${company.name} to their roster.`);
+  }
+
+  getAllCompanies() {
+    console.log(`The companies of ${this.name} are: `);
+    this.#companies.forEach((company, index) => {
+      console.log(`${this.#companies[index].name}`);
+    });
+  }
+}
+
+class Company extends Umbrella {
+  name;
+  #sites = [];
+
+  constructor(name) {
+    super();
+    this.name = name;
+  }
+
+  addSite(site) {
+    this.#sites.push(site);
+    console.log(`${this.name} has added ${site.name} as a Site.`);
+  }
+
+  getNumSites() {
+    console.log(`${this.name} has ${this.#sites.length} sites.`);
+  }
+}
+
+class Site extends Company {
+  name;
+  location;
+  #employees = [];
+
+  constructor(name, location) {
+    super();
+    this.name = name;
+    this.location = location;
+  }
+
+  addEmployee(employee) {
+    this.#employees.push(employee);
+    console.log(`${this.name} has added ${employee.name} to their roster.`);
+  }
 
+  getNumEmployees() {
+    return this.#employees.length;
+  }
+}
 
+class Employee extends Site {
```

Comment:
I think you overdid the inheritance chain, and Employees should be a list of people, rather than being derived from Umbrella